util/pkg/vfs/s3context.go

```diff
@@ -61,6 +64,50 @@ func (s *S3Context) getClient(region string) (*s3.S3, error) {
 	return s3Client, nil
 }
 
+func (s *S3Context) getEndpointClient(region string) (*s3.S3, error) {
+	s.mutex.Lock()
+	defer s.mutex.Unlock()
+
+	s3Client := s.clients[region]
+	if s3Client == nil {
+		Endpoint := os.Getenv("S3_ENDPOINT")
+		if Endpoint == "" {
+			return nil, fmt.Errorf("S3_ENDPOINT is required")
+		}
+		AccessKeyID := os.Getenv("S3_ACCESS_KEY_ID")
+		if AccessKeyID == "" {
+			return nil, fmt.Errorf("S3_ACCESS_KEY_ID cannot be empty when S3_ENDPOINT is not empty")
+		}
+		SecretAccessKey := os.Getenv("S3_SECRET_ACCESS_KEY")
+		if SecretAccessKey == "" {
+			return nil, fmt.Errorf("S3_SECRET_ACCESS_KEY cannot be empty when S3_ENDPOINT is not empty")
+		}
+
+		s3Config := &aws.Config{
+			Credentials:      credentials.NewStaticCredentials(AccessKeyID, SecretAccessKey, ""),
+			Endpoint:         aws.String(Endpoint),
+			Region:           aws.String(region),
+			DisableSSL:       aws.Bool(true),
```

Comment:
We probably should not default to disabling SSL, because otherwise it is hard to make the secure configuration the default later, but I can see how it is the more common case.  I think S3_ENDPOINT can be a full URL (i.e. `http://1.2.3.4/minio`) - do we still need DisableSSL in this case?